src/configSchema.ts

```diff
@@ -0,0 +1,167 @@
+import { Action, LinkType, MatchMode } from "./constants.js";
+import { logger } from "./logger.js";
+import { z } from "zod";
+import ms from "ms";
+
+/**
+ * error messages and map returned upon Zod validation failure
+ */
+enum zodErrorMsg {
+	vercel = "format does not follow vercel's `ms` style ( https://github.com/vercel/ms#examples )",
+	emptyString = "cannot have an empty string",
+	delay = "delay is in seconds, you can't travel back in time.",
+	fuzzySizeThreshold = "fuzzySizeThreshold must be between 0 and 1",
+	injectUrl = "You need to specify rtorrentRpcUrl, transmissionRpcUrl, qbittorrentUrl, or delugeRpcUrl when using 'inject'",
+	dataBased = "Data-Based Matching requires linkType, dataDirs, and linkDir to be defined",
+	riskyRecheckWarn = "It is strongly recommended to not skip rechecking for risky matching mode",
+}
+/**
+ * custom zod error map for logging
+ * @param error ZodIssue messaging object
+ * @param ctx ZodError map
+ * @returns (the custom error for config display)
+ */
+export const ZOD_ERROR_MAP: z.ZodErrorMap = (error, ctx) => {
+	switch (error.code) {
+		case z.ZodIssueCode.invalid_union:
+			return {
+				message: error.unionErrors
+					.reduce((acc, error) => {
+						// @ts-expect-error avoids type conflict string to never
+						error.errors.forEach((x) => acc.push(x.message));
+						return acc;
+					}, [])
+					.join("; "),
+			};
+	}
+
+	return { message: ctx.defaultError };
+};
+/**
+ * helper functions for validation
+ * @return transformed duration (string -> milliseconds)
+ */
+
+function transformDurationString(durationStr: string, ctx) {
+	const duration = ms(durationStr);
+	if (isNaN(duration)) {
+		// adds the error to the Zod Issues
+		ctx.addIssue({
+			code: "custom",
+			message: zodErrorMsg.vercel,
+		});
+	}
+	return duration;
+}
+
+/**
+ * an object of the zod schema
+ * each are named after what they are intended to validate
+ */
+
+export const VALIDATION_SCHEMA = z
+	.object({
+		delay: z
+			.number()
+			.positive({
+				message: zodErrorMsg.delay,
+			})
+			.default(10),
```

Comment:
So my understanding was that your "defaults" using fallback in commander would accept any value provided, and only fallback when otherwise undefined. This defaults upon erroneous values being provided and not valid.

For instance, with delay - if someone was to provide `"10s"` rather than `10` - it would default to 10. I guess this could be better utilized with error'ing, but this was more of an afterthought during schema-creation and I didn't put much thought into the consequences of defaulting.

I don't really have a preference on this implementation being kept, it was merely just replicating what I understood at the time about the defaults, and using that in the schema.

Will wait for your thoughts.

Reply:
So do you want me to remove all "failing to a default" behavior? I assume with Zod we want pretty strict config validation, so that seems like the approach you're insinuating. 

Throw and terminate on anything wrong with the config basically.

Reply:
Yeah, I think thats the best way to go. by sticking with the `fallback` pattern we've already got for now, I think we're able to keep the door open to using `.default` later on. If we _do_ decide to do that later on, I would like to fully embrace it for all options.

Reply:
gonna copy your other message to this thread to keep things centralized: 
> To elaborate, apiAuth I think we are forcing right? So that needs to be addressed as well (remove the option and force true, I can just do that with Zod I think or we can hardcode it. The thing with verbose is that it's not in the config template by default - so we'd be relying on fallback in commander and just strictly forcing a boolean.

thats a good catch about verbose - I hadn't realized it's not configurable via config.js. For API auth I'm still not sure whether I want to make it opt-out for existing users by changing the last `fallback` arg from `false` to `true`, or whether we really should make it non negotiable. We can chat in #development about that 🙂

Reply:
Understood, will keep this unresolved for the time being until we discuss it.

Reply:
As a sidenote, I think we could just add `verbose: false` to the template?

Reply:
We'd have to add it to the fallback call as well, but I don't want to make this PR any bigger